File: oracle.py
import random


class Oracle:
    def __init__(self, alpha_range, delta_range):
        self._alpha_n = random.randint(*alpha_range)
        self._alpha_0 = self._alpha_n
        self._delta = random.randint(*delta_range)

    def ask(self, alpha, delta):
        if alpha == self._alpha_n and delta == self._delta:
            return True

        self._alpha_n += self._delta
        return False

    def reset(self):
        self._alpha_n = self._alpha_0

    def get_secrets(self):
        return self._alpha_0, self._delta

File: dart/oracle.dart
import 'dart:math';

class Oracle {
  int alphaCurrent, alphaSecret, deltaSecret;

  Oracle(alphaMin, alphaMax, deltaMin, deltaMax) {
    var rng = new Random();
    this.alphaSecret = rng.nextInt(alphaMax) + alphaMin;
    this.alphaCurrent = this.alphaSecret;
    this.deltaSecret = rng.nextInt(deltaMax) + deltaMin;
  }

  bool ask(int alpha, int delta) {
    if (alpha == this.alphaCurrent && delta == this.deltaSecret)
      return true;

    this.alphaCurrent += this.deltaSecret;
    return false;
  }

  void reset() {
    this.alphaCurrent = this.alphaSecret;
  }

  OracleSecrets getSecrets() {
    return OracleSecrets(this.alphaSecret, this.deltaSecret);
  }
}

class OracleSecrets {
  int alpha, delta;

  OracleSecrets(this.alpha, this.delta) {}
}